tests/posix/env/testcase.yaml

```diff
@@ -0,0 +1,32 @@
+common:
+  filter: not (CONFIG_NATIVE_BUILD and CONFIG_EXTERNAL_LIBC)
+  arch_exclude:
+    - posix
```

Comment:
Instead of excluding platforms that build picolibc from source, it would be better to undefine & define the macro in the .c file.
(Otherwise it is not just native_sim when building with picolibc but anybody building picolibc from source until keith's pr is merged).

Reply:
I will probably not be defining in the source for reasons already specified, but in the long run, it should be perfectly fine to do that in source files.